packages/inter-protocol/src/clientSupport.js

```diff
@@ -0,0 +1,228 @@
+// @ts-check
+
+import { Fail } from '@agoric/assert';
+
+// XXX support other decimal places
+const COSMOS_UNIT = 1_000_000n;
+
+// NB: not really a Proposal because the brands are not remotes
+// Instead they're copyRecord like  "{"boardId":"board0257","iface":"Alleged: IST brand"}" to pass through the boardId
+// mustMatch(harden(proposal), ProposalShape);
+/**
+ * Give/want
+ *
+ * @param {Record<string, import('@agoric/vats/tools/board-utils').BoardRemote>} brands
+ * @param {{ giveMinted?: number, wantMinted?: number } | { collateralBrandKey: string, giveCollateral?: number, wantCollateral?: number }} opts
+ * @returns {Proposal}
+ */
+const makeVaultProposal = (brands, opts) => {
+  const proposal = { give: {}, want: {} };
+
+  if ('giveCollateral' in opts && opts.giveCollateral) {
+    const { collateralBrandKey } = opts;
+    proposal.give.Collateral = {
+      brand: brands[collateralBrandKey],
```

Comment:
the cli seems to supply `collateralBrand`, not `collateralBrandKey`. I'm struggling to get this code to work. It works if I change to `collateralBrand`. But maybe it would be better to change the CLI?

Reply:
Better to change the CLI, yes https://github.com/Agoric/agoric-sdk/pull/7133

It now has a little more type checking. I would normally test the CLI before merging it but I'm juggling a lot of things. If you already are using the CLI, mind grabbing that commit and seeing whether the problem is gone?